codeinsights-db/conf/postgresql.conf

```diff
@@ -0,0 +1,91 @@
+# -----------------------------
+# PostgreSQL configuration file
+# -----------------------------
+#
+# This file consists of lines of the form:
+#
+#   name = value
+#
+# (The "=" is optional.)  Whitespace may be used.  Comments are introduced with
+# "#" anywhere on a line.  The complete list of parameter names and allowed
+# values can be found in the PostgreSQL documentation.
+#
+# The commented-out settings shown in this file represent the default values.
+# Re-commenting a setting is NOT sufficient to revert it to the default value;
+# you need to reload the server.
+#
+# This file is read on server startup and when the server receives a SIGHUP
+# signal.  If you edit the file on a running system, you have to SIGHUP the
+# server for the changes to take effect, run "pg_ctl reload", or execute
+# "SELECT pg_reload_conf()".  Some parameters, which are marked below,
+# require a server shutdown and restart to take effect.
+#
+# Any parameter can also be given as a command-line option to the server, e.g.,
+# "postgres -c log_connections=on".  Some parameters can be changed at run time
+# with the "SET" SQL command.
+#
+# Memory units:  kB = kilobytes        Time units:  ms  = milliseconds
+#                MB = megabytes                     s   = seconds
+#                GB = gigabytes                     min = minutes
+#                TB = terabytes                     h   = hours
+#                                                   d   = days
+
+#------------------------------------------------------------------------------
+# POSTGRESQL DEFAULT CONFIGURATION
+#------------------------------------------------------------------------------
+
+# Below is PostgreSQL default configuration.
+# You should apply your own customization in the CUSTOMIZED OPTIONS section below
+# to avoid merge conflict in the future.
+
+listen_addresses = '*'
+max_connections = 100
+shared_buffers = 128MB
+dynamic_shared_memory_type = posix
+max_wal_size = 1GB
+min_wal_size = 80MB
+log_timezone = 'UTC'
+datestyle = 'iso, mdy'
+timezone = 'UTC'
+lc_messages = 'en_US.utf8'
+lc_monetary = 'en_US.utf8'
+lc_numeric = 'en_US.utf8'
+lc_time = 'en_US.utf8'
+default_text_search_config = 'pg_catalog.english'
+
+#------------------------------------------------------------------------------
+# SOURCEGRAPH RECOMMENDED OPTIONS
+#------------------------------------------------------------------------------
+
+# Below is Sourcegraph recommended Postgres configuration.
+# You should apply your own customization in the CUSTOMIZED OPTIONS section below
+# to avoid merge conflict in the future.
+#
+# NOTES: Options below is based on the default resource configuration of `codeinsights-db` container.
+# Depending on your environment, you may need to adjust the settings in CUSTOMIZED OPTIONS below
+
+shared_buffers = 509546kB
+work_mem = 3184kB
+maintenance_work_mem = 254773kB
+effective_io_concurrency = 200
+max_worker_processes = 19
```

Comment:
Oh boy - I started looking into this because I thought 19 was such a weird number, and it turns out this is the [initial configuration](https://github.com/sourcegraph/deploy-sourcegraph/pull/1851) from Feb 2021 and tuned for timescaledb:
```
The postgres.conf included here comes from TimescaleDB itself, I included comments in the ConfigMap about differences and how to use timescaledb-tune to generate a good config file.
```

So, now that we're using vanilla Postgres, these numbers might not be as relevant. I think it's ok for this PR to keep those defaults but we should ask the Code Insights team to review the settings soon.

Reply:
Asked Code Insights team to investigate here: https://sourcegraph.slack.com/archives/C014ZCKMCAV/p1649421612224539